### PR TITLE
WIP: region choice

### PR DESCRIPTION
**Pull request by abdulajet**
adds a `--apiRegion` flag to the `config:set` command and then uses computed values for the base URLs

### CODE REVIEW

**Review by dragonmantank — CHANGES_REQUESTED** [review]
I think this mostly fixes it for conversations, which covers the original ticket, but we have this problem with Voice as well.

And looking back at the [docs](https://developer.vonage.com/voice/voice-api/guides/troubleshooting/node), I think we need to be specific to the actual data center. We'll also need to make this expandable to handle video in the future, which is a third URL. What I would do is:

1. Add `--apiHost`, `--restHost`, and `--videoHost` flags to replace `--apiRegion`. These take arbitrary strings
2. Augment the [client creation](https://github.com/Vonage/vonage-cli/blob/92c72bfb87f884e0ce9078409e78e803f9610f84/packages/utils/src/index.ts#L76) to pass these additional options alongside the user agent append

This should let us use the already existing overrides with all the APIs. The onus is on the user to pass a valid URL, but we already require that in the SDKs today so a user overriding this should already be familiar with what they need to pass. This will also make it easier when we swap out Node SDK v3 under the hood, as it will take these same options.

I think your solution is the right direction, we can just take advantage of some of the existing code.

**Comment by abdulajet** [comment]
> I think this mostly fixes it for conversations, which covers the original ticket, but we have this problem with Voice as well.

yes sorry, didn't wanna do everything without getting some feedback. I'll implement your suggestions

File: packages/utils/src/index.ts [hunk 1]
```diff
@@ -73,7 +58,22 @@ export default abstract class BaseCommand<T extends typeof BaseCommand.flags> ex
...
+
+        if (this._restHost) {
+            options = { ...options, ...{ restHost: this._restHost} }
+        }
+
+        if (this._videoHost) {
+            options = { ...options, ...{ videoHost: this._videoHost} }
+        }
+
+        console.log(options)
```
Comment (dragonmantank) [inline packages/utils/src/index.ts:75]: I would take this out when we write the tests :P

**Review by dragonmantank — CHANGES_REQUESTED** [review]
Looks good, other than the `console.log()` that got left in. Once tests are in I think we're good.

**Comment by manchuck** [comment]
Closing this PR as region selection will be added with the upgrade to V3 of the SDK.

**Pull request closed without merging**